tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs

```diff
@@ -315,16 +315,41 @@ private static T GetDelegateForNativeFunction<T>(IntPtr handle, string functionN
 
         private static List<string> GetDatadogNativeFolders(FrameworkDescription frameworkDescription, string runtimeId)
         {
-            // first get anything "home folder" like, then combine with the profiler's location
-            // taking into account that this should be the same place
+            // first get anything "home folder" like
+            // then get the program files folder, if running under Windows
+            // then combine with the profiler's location
+            // taking into account that these locations could be the same place
 
             var paths = GetHomeFolders(runtimeId);
+
+            if (frameworkDescription.OSPlatform == OSPlatform.Windows)
+            {
+                var programFilesFolder = GetProgramFilesFolder();
+                paths.Add(programFilesFolder);
+            }
+
             var profilerFolder = GetProfilerFolder(frameworkDescription);
-            paths.Add(profilerFolder);
+            if (!string.IsNullOrWhiteSpace(profilerFolder))
+            {
+                paths.Add(profilerFolder);
+            }
+            else
+            {
+                // this is expected under Windows, but problematic under other OSs
+                Log.Debug("Couldn't find profilerFolder");
+            }
 
             return paths.Distinct().ToList();
         }
 
+        private static string GetProgramFilesFolder()
+        {
+            // should be already adapted to the type of process / OS
+            var programFilesRoot = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
```

Comment:
Note that the tracer is not always installed into the Program Files folder. Users are allowed to change install location in the msi, and we have several who specifically don't want us to install anything into C:.

Side note: there is a known issue where the x86 dll _always_ goes into `C:\Program Files (x86)` (or wherever the equivalent is on their system) even if they change the install path. All other files (the x64 dll, managed files, json, etc) respect the user selection.

Reply:
Maybe you can use the registry key we set during installation:
```
Keys: "HKLM\Software\Datadog\Datadog .NET Tracer 64-bit" (when using the x64 msi)
      "HKLM\Software\Datadog\Datadog .NET Tracer 32-bit" (when using the x86 msi)
Name: "InstallPath"
Value="[INSTALLFOLDER]"
```

https://github.com/DataDog/dd-trace-dotnet/blob/86c0bc5542afa42600c39931f741f3b1f7b4ddf2/tracer/src/WindowsInstaller/Product.wxs#L181-L182

Reply:
We're trying to fix the x86 on x64 case, which is the one with the MSI bug, so I think we do want to check the default location. I've tested the x86 on x64 case manually with a modified install location and the code from the first commit.

I think in a lot of cases, when the DD library is running with the same bitness as the OS, we'll be able to find ddwaf.dll with the environment variables DD_DOTNET_TRACER_HOME and DD_INTEGRATIONS, which are also set by the MSI.

I've added 81533d7f which reads the install location from the registry, as suggested, this could prove useful if the user has customized there setup and changed DD_DOTNET_TRACER_HOME or DD_INTEGRATIONS.

Reply:
FYI: DD_INTEGRATIONS will be no longer necessary for 2.0, so we should avoid relying on this.

Reply:
I wouldn't say we rely on DD_INTEGRATIONS, it's more it's one of a number of sources of information we use in our library finding heuristic.

Reply:
> We're trying to fix the x86 on x64 case, which is the one with the MSI bug, so I think we do want to check the default location.

Thanks for adding the registry check. We shouldn't fix one issue (finding the x86 dll) by relying on another issue (installing into the wrong path). The registry fix should keep working even after we fix the msi bug. 👍🏽 

Maybe someday we can have the msi install them under `DD_DOTNET_TRACER_HOME/{architecture}` like we do with some of the other packages...

Reply:
Looks like you have all the bases covered now. Resolving this comment.